### PR TITLE
Add orderbook docs

### PR DESCRIPTION
**Pull request by prettymuchbryce**
Adds documentation for the `/dex` orderbook APIs.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+|----------|---------|-----------|
+|makerMarket|string|The Solo [market](https://docs.dydx.exchange/#/overview?id=markets) of the Maker amount|
+|takerMarket|string|The Solo [market](https://docs.dydx.exchange/#/overview?id=markets) of the Taker amount|
+|makerAmount|string|The amount of token the Maker is offering in base units|
+|takerAmount|string|The amount of token the Maker is requesting from the taker base units|
+|makerAccountOwner|string|The Ethereum address of the Maker.|
+|takerAccountOwner|string|The Ethereum address of the Taker.|
+|makerAccountNumber|string|The Solo [account number](https://docs.dydx.exchange/#/overview?id=markets) of the Maker|
+|takerAccountNumber|string|The Solo [account number](https://docs.dydx.exchange/#/overview?id=markets) of the Taker|
+|expiration|string|The time in unix seconds at which this order will be expired and can no longer be filled.|
```
Comment (BrendanChou) [inline docs/api.md]: May want to specify that 0 expiration is no expiration

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+|order|Object|A valid signed order JSON object|
+|fillOrKill|boolean|Whether or not this order is a market order.|
+ 
+note: Market orders execute immediately and no part of the market order will go on the open order
+book. Market orders will either be completely filled, or not filled. Partial fills are not possible.
+
+Example Request Body:
+```JSON
+{
+	"fillOrKill" true,
```
Comment (BrendanChou) [inline docs/api.md]: missing colon

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+__Order fields__
+
+|Field Name|JSON type|Description|
+|----------|---------|-----------|
+|makerMarket|string|The Solo [market](https://docs.dydx.exchange/#/overview?id=markets) of the Maker amount|
+|takerMarket|string|The Solo [market](https://docs.dydx.exchange/#/overview?id=markets) of the Taker amount|
+|makerAmount|string|The amount of token the Maker is offering in base units|
+|takerAmount|string|The amount of token the Maker is requesting from the taker base units|
+|makerAccountOwner|string|The Ethereum address of the Maker.|
+|takerAccountOwner|string|The Ethereum address of the Taker.|
```
Comment (AntonioJuliano) [inline docs/api.md]: add a note that for our API this must be set to our account (same for taker account number)

File: docs/api.md [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+```
+Content-Type: application/json
+```
+
+Request Body:
+
+|Field Name|JSON type|Description|
+|----------|---------|-----------|
+|order|Object|A valid signed order JSON object|
+|fillOrKill|boolean|Whether or not this order is a market order.|
```
Comment (AntonioJuliano) [inline docs/api.md]: A better description might be something like: Whether the order should be canceled if it cannot be immediately filled

(having a fill or kill limit order also makes sense)

File: docs/api.md [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+Returns:
+`201` if successful
+
+### DELETE /v1/dex/orders/:hash
+
+Description:
+Cancels an open order by hash.
+
+Please note you will need to provide a valid cancellation signature in order to cancel an order.
```
Comment (AntonioJuliano) [inline docs/api.md]: put some info as to what you have to sign maybe

File: docs/api.md (hunk 3 above)
Comment (prettymuchbryce) [reply docs/api.md]: Do we know what this owner/number is yet?

**Pull request merged**